Feature ft232 h mac and linux support

bryancostanich opened the pull request:
I've only tested mac support. though Linux is nominally in.

Also, only tested I2C, but that's working.

make sure to test on Windows to make sure i haven't broken anythig.

bryancostanich committed:
initial attempt at mac + linux support for the FT232H

bryancostanich committed:
added note about the MCP driver

bryancostanich committed:
er, VCP.

bryancostanich committed:
lots of work to get the FT232H to actually enumerate and connect on mac.

bryancostanich committed:
fix up outdated note on VCP working

bryancostanich committed:
uncommented I2C tests

adrianstevens commented on Source/Meadow.Foundation.Peripherals/ICs.IOExpanders.Ftxxxx/Samples/Ft232h_Sample/Program.cs:
```diff
@@ -104,40 +138,51 @@ async Task TestI2C(FtdiExpander expander)
-    }
-
-    var s = false;
+// async Task TestGpio(IEnumerable<FtdiExpander> expanders)
```
does this need to be commented out?

bryancostanich replied:
nah. it can be uncommented.

bryancostanich committed:
uncommented some test methods per PR review

bryancostanich committed:
renamed MPSSE I2C and SPI implementations

currently, the MPSEE I2C and SPI bus classes are named `Ft232hSpiBus` and `Ft232hI2C` bus, but they're actually shared across various "H" chips like the FT2232H, FT232H, and FT4232 or whatever. so this makes them more understandable and fits their implementation better.

ctacke committed:
fix: FTD2xx test update

ctacke committed:
Merge branch 'develop' into feature-FT232H_Mac_and_Linux_Support
